### PR TITLE
feat: implemented login auth workflow

### PR DESCRIPTION
**Pull request by Quwaysim**
Implemented Login Auth Workflow.
Implemented the OTP view and logic.
Implemented Stacked Arch approach for form handling in signup, otp and login views .

Screenshot_1631959883

Screenshot_1631987022

### CODE REVIEW

**Review by mariamadebolahamzat — APPROVED** [review]
All changes made are good and can be merged.

File: lib/ui/view/login/login_viewmodel.dart [hunk 1]
```diff
@@ -43,12 +41,16 @@ class LoginViewModel extends BaseViewModel {
     const endpoint = '/auth/login';
     if (email.text == '' || password.text == '') {
       loading(false);
+      //Hides the keyboard for the failure snackbar to be visible
+      FocusScope.of(context).unfocus();
```
Comment (Blazebrain) [inline lib/ui/view/login/login_viewmodel.dart:45]: Remove flutter/material import and adjust to fit.

**Review by Blazebrain — CHANGES_REQUESTED** [review]
Good to go once changes requested have bee fixed

File: lib/ui/view/login/login_viewmodel.dart (hunk 1 above)
Comment (Quwaysim) [reply lib/ui/view/login/login_viewmodel.dart]: changes made

**Review by Blazebrain — APPROVED** [review]
Good to go

**Review by mariamadebolahamzat — APPROVED** [review]

**Pull request merged**